python/test_dcr_template.py

```diff
@@ -0,0 +1,629 @@
+        # Also check that the generator is now exhausted
+        with self.assertRaises(StopIteration):
+            next(dcr_gen)
+
+    def test_dcr_values(self):
+        """Check DCR against pre-computed values."""
+        azimuth = 0.0
+        elevation = 50.0
+        dcr_ref_vals = [(1.9847367904770623, 1.6467981843302726),
+                        (1.6467981843302726, 1.3341803407311699),
+                        (1.3341803407311699, 1.0443731947908652),
+                        (1.0443731947908652, 0.77517513542339489),
+                        (0.77517513542339489, 0.52464791969238367),
+                        (0.52464791969238367, 0.29107920440002155),
+                        (0.29107920440002155, 0.072951238300825172),
+                        (0.072951238300825172, -0.13108543143740825),
+                        (-0.13108543143740825, -0.3222341473268886),
+                        (-0.3222341473268886, -0.50157094957602733),
+                        (-0.50157094957602733, -0.6700605866796161),
+                        (-0.6700605866796161, -0.8285701993878597),
+                        (-0.8285701993878597, -0.97788106062563773),
+                        (-0.97788106062563773, -1.1186986838806061),
+                        (-1.1186986838806061, -1.2125619138571659)]
+        bp = self.bandpass
+        dcr_gen = DcrModel._dcr_generator(bp, pixel_scale=self.pixel_scale, elevation=elevation,
+                                          azimuth=azimuth, use_midpoint=False)
+        n_step = int(np.ceil((bp.wavelen_max - bp.wavelen_min) / bp.wavelen_step))
+        for f in range(n_step):
+            dcr = next(dcr_gen)
+            self.assertFloatsEqual(dcr.dy.start, dcr_ref_vals[f][0])
+            self.assertFloatsEqual(dcr.dy.end, dcr_ref_vals[f][1])
+
+
+class BandpassTestCase(lsst.utils.tests.TestCase):
+    """Tests of the interface to Bandpass from lsst.sims.photUtils."""
+
+    def setUp(self):
+        """Define parameters used by every test."""
+        self.band_name = 'g'
+        self.wavelength_step = 10
+        self.bandpass = DcrModel._load_bandpass(band_name=self.band_name,
+                                                wavelength_step=self.wavelength_step)
+
+    def test_step_bandpass(self):
+        """Check that the bandpass has necessary methods, and those return the correct number of values."""
+        bp = self.bandpass
+        bp_wavelen, bandpass_vals = bp.getBandpass()
+        n_step = int(np.ceil((bp.wavelen_max - bp.wavelen_min) / bp.wavelen_step))
+        self.assertEqual(n_step + 1, len(bandpass_vals))
+
+
+class DcrModelTestBase(lsst.utils.tests.TestCase):
+
+    def setUp(self):
+        band_name = 'g'
+        n_step = 3
+        pixel_scale = 0.25
+        self.kernel_size = 5
+        self.size = 20
+        # NOTE that this array is randomly generated for each instance.
+        self.array = np.random.random(size=(self.size, self.size))
```

Comment:
The seed value is NOT set for this randomly generated array. I will add to the comment to call this out explicitly.